metadata/server/server.go

```diff
@@ -6,38 +6,49 @@ package main
 
 import (
 	"fmt"
+	"github.com/featureform/metadata/search"
 	"os"
 
 	help "github.com/featureform/helpers"
 	"github.com/featureform/metadata"
-	"github.com/featureform/metadata/search"
 	"go.uber.org/zap"
 )
 
 func main() {
 
 	etcdHost := help.GetEnv("ETCD_HOST", "localhost")
-	etcdPort := help.GetEnv("ETCD_PORT", "6379")
+	etcdPort := help.GetEnv("ETCD_PORT", "2379")
 	logger := zap.NewExample().Sugar()
 	addr := help.GetEnv("METADATA_PORT", "8080")
+	disableTypesense := help.GetEnv("DISABLE_TYPESENSE", "false")
 	storageProvider := metadata.EtcdStorageProvider{
 		metadata.EtcdConfig{
 			Nodes: []metadata.EtcdNode{
 				{etcdHost, etcdPort},
 			},
 		},
 	}
-	fmt.Println("TS Port", os.Getenv("TYPESENSE_PORT"), "TS HOST", os.Getenv("TYPESENSE_HOST"), "TS KEY", os.Getenv("TYPESENSE_APIKEY"))
-	config := &metadata.Config{
-		Logger:  logger,
-		Address: fmt.Sprintf(":%s", addr),
-		TypeSenseParams: &search.TypeSenseParams{
-			Port:   help.GetEnv("TYPESENSE_PORT", "8108"),
-			Host:   help.GetEnv("TYPESENSE_HOST", "localhost"),
-			ApiKey: help.GetEnv("TYPESENSE_APIKEY", "xyz"),
-		},
-		StorageProvider: storageProvider,
+	var config *metadata.Config
+	if disableTypesense == "false" {
```

Comment:
config = &metadata.Config{
			Logger:          logger,
			Address:         fmt.Sprintf(":%s", addr),
			StorageProvider: storageProvider,
		}
repeats in either block. Should it be declared before and then TypeSennseParams can just be added in a single if block after to the config?

Reply:
removed duplicate code